src/Compilers/VisualBasic/Portable/Lowering/ExpressionLambdaRewriter/ExpressionLambdaRewriter_BinaryOperator.vb

```diff
@@ -317,34 +391,40 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                          opKind = BinaryOperatorKind.Power OrElse opKind = BinaryOperatorKind.Multiply OrElse
                          opKind = BinaryOperatorKind.Add OrElse opKind = BinaryOperatorKind.Subtract OrElse
                          opKind = BinaryOperatorKind.Divide OrElse opKind = BinaryOperatorKind.IntegerDivide OrElse
-                         opKind = BinaryOperatorKind.Modulo OrElse opKind = BinaryOperatorKind.Concatenate OrElse
+                         opKind = BinaryOperatorKind.Modulo OrElse
                          opKind = BinaryOperatorKind.LeftShift OrElse opKind = BinaryOperatorKind.RightShift)
             Debug.Assert((node.OperatorKind And BinaryOperatorKind.UserDefined) = 0)
 
             ' Do we need to use special helpers?
             Dim helper As MethodSymbol = Nothing
             If resultUnderlyingSpecialType = SpecialType.System_Object Then
                 helper = GetHelperForObjectBinaryOperation(opKind)
-
+                If helper Is Nothing Then ' Don't know how to do 'BinaryOperatorKind.Power' without the method
+                    Return _factory.BadExpression(Visit(node.Left), Visit(node.Right))
+                End If
             ElseIf resultUnderlyingSpecialType = SpecialType.System_Decimal Then
                 helper = GetHelperForDecimalBinaryOperation(opKind)
-
-            ElseIf opKind = BinaryOperatorKind.Concatenate Then
-                helper = DirectCast(_factory.SpecialMember(SpecialMember.System_String__ConcatStringString), MethodSymbol)
-
+                If helper Is Nothing Then ' Don't know how to do 'BinaryOperatorKind.Power' without the method
```

Comment:
> This comment and the one above look inaccurate. We might not be doing 'Power' operation.

I think the comment accurately reflects the motivation for adding the check. It is an error to fail to find the helper for any operator. The question is what the rewriter should do to recover. The error has been reported, but the rewriter needs to produce a bound node to return. It used to be that the rewriter would continue as though the helper is not needed and would produce a node of that shape, there are expression tree factories to do that (despite the fact that that wouldn't accurately reflect semantics of the operator). In order to maintain that recovery strategy, we would have to make use of a "Power" factory method that is never used in a success scenario, which, in my opinion, would lead to a confusion of anyone who would want to figure out what is a minimal set of factories required by VB. The new error recovery approach is going to work well for all operators, but the primary motivation is to exclude usage of a "Power" factory method that would never be needed in a success scenario.